### PR TITLE
Add "test" and "checks" scripts and pre-commit hook

### PR DESCRIPTION
**Pull request by edavey**
This PR adds

- a script to run all the pipeline tasks with `script/test` or `npm run test:full`
- a script to run just the i) linting and ii) typechecking tasks with `script/checks` or `npm run checks`
- an npm script to generate types from the API with `npm run generate-types`
- adds a pre-commit hook to run the 'checks' (easily disabled) e.g. by passing `env HUSKY=0 inline`

### CODE REVIEW

**Review by patrickjfl — DISMISSED** [review]
Looks great, nice job! I can imagine running the whole suite of unit and integration tests as a pre commit hook might get a bit much down the line, but it's really cool anyway 👍

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
[the remark this answers is not in the document]
Comment (edavey) [reply .husky/pre-commit:2]: would it be better if the pre-commit hook only ran the typechecking and the linting maybe?

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "security_audit": "npx audit-ci --config audit-ci.json",
     "int-test": "cypress run --config video=false",
     "int-test-ui": "cypress open",
+    "generate-types": "script/generate-types",
```
Comment (Harriethw) [inline package.json:30]: nice!

File: .husky/pre-commit (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harriethw) [reply .husky/pre-commit:2]: I think I'd prefer that, as I run the tests regularly myself, but don't typecheck or lint before pushing usually. And waiting more than a few seconds to commit something would be a pain.

[the remark this answers is not in the document]
Comment (edavey) [reply .husky/pre-commit:2]: thanks both for your input. I've reworked based on your comments...

File: script/test [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+set -e
+
+echo "==> Linting the code"
+
+npm run lint
+
+echo "==> Pulling the latest type from the API repo"
+
+npm run generate-types
```
[the remark this answers is not in the document]
Comment (edavey) [reply script/test]: yes, I think you're right about this. I just copied the `test` script from CAS1, but you're right that adding in the types in a pre-commit scenario isn't that helpful!

**Comment by edavey** [comment]
@patrickjfl, @Harriethw 

Could you take another look now? I've: 

- added `npm run security_audit` to the `test:full` script and 
- added a `checks` script with just the typechecking and linting for use in a pre-commit hook

**Review by Harriethw — APPROVED** [review]
Thanks for this @edavey 👍

**Pull request merged**